source/guides/stacks/hpcrs.rst

```diff
@@ -0,0 +1,225 @@
+      rm go$VERSION.$OS-$ARCH.tar.gz
+
+#. Setup the environment for Go
+
+   .. code-block:: bash
+
+      echo 'export GOPATH=${HOME}/go' >> ~/.bashrc && \
+      echo 'export PATH=/usr/local/go/bin:${PATH}:${GOPATH}/bin' >> ~/.bashrc && \
+      source ~/.bashrc
+
+
+#. Install :command:`dep` for dependency resolution with Singularity v3.0.0
+
+   .. code-block:: bash
+
+      go get -u github.com/golang/dep/cmd/dep
+
+#. Download Singularity
+
+   .. code-block:: bash
+
+      go get -d github.com/sylabs/singularity
+
+
+   .. note::
+
+      Go will complain that there are no Go files, but it will still download the Singularity source code to the appropriate directory within the $GOPATH.
+
+#. Checkout version 3.0.0 of Singularity
+
+   .. code-block:: bash
+
+      export VERSION=v3.0.3 # or another tag or branch if you like && \
+      cd $GOPATH/src/github.com/sylabs/singularity && \
+      git fetch && \
+      git checkout $VERSION # omit this command to install the latest bleeding edge code from master
+
+
+#. Build Singularity
+
+   Singularity uses a custom build system called makeit. mconfig is called to generate a Makefile and then make is used to compile and install.
+   The devpkg-openssl, devpkg-util-linux  package may be required and can be installed using the :command:`sudo swupd bundle-add <pkg-name>`
+
+   .. code-block:: bash
+
+      ./mconfig && \
+      make -C ./builddir && \
+      sudo make -C ./builddir install
+
+
+#. Configure bash completion (optional)
+
+   To enjoy bash completion with Singularity commands and options, source the bash completion file. Add this command to your ~/.bashrc file so that bash completion continues to work in new shells
+
+   .. code-block:: bash
+
+      . /usr/local/etc/bash_completion.d/singularity
+
+
+Converting Docker images to Singularity Images
+**********************************************
+
+#. Download d2s
+
+   :command:`d2s` os an open source tool to convert Docker images to Singularity images. You can use the script in the location where it is downloaded, or install it using the included :file:`setup.py` file with the :command:`python setup.py install`
+
+
+   .. code-block:: bash
+
+      git clone https://github.com/intel/stacks.git
+      cd hpcrs/d2s
+
+#. List local Docker images
+
+   .. code-block:: bash
+
+      python d2s.py --list_docker_images
+
+   Your output can appear like this:
+
+   .. code-block:: console
+
+      ==============================
+      Docker images present locally
+      ==============================
+      ID         NAME
+      0: clearlinux/stacks-dlrs-mkl
+      1: clearlinux/stacks-dlrs_2-mkl
+      ==============================
+
+#. Convert to Singularity images
+
+   To convert the Docker images to Singularity images, use the :command:`d2s` script with the ID numbers of the images you wish to convert.
+
+
+   .. code-block:: bash
+
+      python d2s.py --convert_docker_images <ID_1> <ID_2>
+
```

Comment:
In my test, I used the ID for `sysstacks/dbrs-memcached-clearlinux`. 
The conversion worked fine. But running the converted file produced a failure: 

```
FATAL:   container creation failed: mount error: can't mount image /proc/self/fd/8: failed to mount squashfs filesystem: invalid argument
```

When I converted the file, I also noticed this warning: 
`WARNING: Authentication token file not found : Only pulls of public images will succeed`

Could this  be the root cause?


I found similar issues to this on `sylabs`: 
https://github.com/sylabs/singularity/issues/2441
I tried clearing the cache too, but this didn't have an effect: 
https://github.com/sylabs/singularity/issues/2625#issuecomment-459299781